### DISCUSSION TITLE
How do I migrate a field from one type to another?

### DISCUSSION THREAD

**Discussion by bearcatsandor** [beetbox/beets#4681] [Q&A]
I have a custom field called "rating" that I have a value of 1 - 100.  I used to be able to do things like `beet ls 菅野よう子 rating:90..100`
and that would work. Then one day it stopped working. 

Looking through my configs I saw that I had commented out
`types:
    rating: int`

So half of my ratings weren't calculating. However, if I uncomment that some of my ratings come out as 0 when listed because they are actually strings.

Is there an easy way to convert these ratings to ints so that they display and resolve properly?

**Comment by sampsyo** [comment]
Hmm; that's interesting. I actually would have guessed that this would "just work," i.e., changing the configuration would have sufficed and beets would reinterpret the existing data in a numeric way.

Can you include an example of a file or two (using `beet ls`) where the value looks like a number but is actually a string? Is there any chance that those are not actually purely numeric, and they have whitespace around them or something?

**Comment by bearcatsandor** [comment]
Here's what the album Stand Alone Complex 3 looks like when my ratings are set to int:

菅野よう子 - Scott Matthew - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - the end of all you'll know - 2:21 - 2.0 channel - $rating
菅野よう子 - Gabriela Robin - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - トルキア - 6:08 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - know your enemy - 6:24 - 2.0 channel - 0
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - レーザーシーカー - 3:14 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - break through - 2:36 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - flying low - 4:36 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - エウロペ - 4:31 - 2.0 channel - 0
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - 半島の東 - 4:06 - 2.0 channel - 0
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - 未完成ラブストーリー - 1:54 - 2.0 channel - 0
菅野よう子 - Ilaria Graziano - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - CHRisTmas in the SiLenT ForeSt - 6:55 - 2.0 channel - 0
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - access all areas - 1:35 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - sacred terrorist - 4:04 - 2.0 channel - 0
菅野よう子 - Scott Matthew - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - dear john - 5:01 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - 35.7℃ - 1:47 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - スマイル - 4:11 - 2.0 channel - $rating
菅野よう子 - ORIGA feat. Ben Del Maestro - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - flashback memory plug - 1:09 - 2.0 channel - $rating
菅野よう子 - Ilaria Graziano - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - dew - 5:49 - 2.0 channel - 0

Here is is again without the config lines that set rating to int:

菅野よう子 - Scott Matthew - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - the end of all you'll know - 2:21 - 2.0 channel - $rating
菅野よう子 - Gabriela Robin - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - トルキア - 6:08 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - know your enemy - 6:24 - 2.0 channel - 98.0
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - レーザーシーカー - 3:14 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - break through - 2:36 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - flying low - 4:36 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - エウロペ - 4:31 - 2.0 channel - 100.0
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - 半島の東 - 4:06 - 2.0 channel - 98.0
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - 未完成ラブストーリー - 1:54 - 2.0 channel - 99.0
菅野よう子 - Ilaria Graziano - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - CHRisTmas in the SiLenT ForeSt - 6:55 - 2.0 channel - 81.0
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - access all areas - 1:35 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - sacred terrorist - 4:04 - 2.0 channel - 94.0
菅野よう子 - Scott Matthew - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - dear john - 5:01 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - 35.7℃ - 1:47 - 2.0 channel - $rating
菅野よう子 - 菅野よう子 - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - スマイル - 4:11 - 2.0 channel - $rating
菅野よう子 - ORIGA feat. Ben Del Maestro - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - flashback memory plug - 1:09 - 2.0 channel - $rating
菅野よう子 - Ilaria Graziano - 2005 - 攻殻機動隊 STAND ALONE COMPLEX O.S.T. 3 - dew - 5:49 - 2.0 channel - 98.0

The last column is the one that matters.

What's really wierd is that if I comment out the lines that turn rating into an int, and do `beet ls album:stand alone 3 rating:80..` it returns nothing.  I mean, yeah, it's a string so no math on it. If i turn ratings to an int and do `beet ls album:stand alone  rating:80..` which should pull all tracks from all of these albums with a rating > 79 I get

菅野よう子 - 菅野よう子 - 2003 - 攻殻機動隊 STAND ALONE COMPLEX be Human - ロッキーはどこ？ - 4:25 - 2.0 channel - 90
菅野よう子 - 菅野よう子 - 2003 - STAND ALONE COMPLEX O.S.T.+ - some other time - 4:07 - 2.0 channel - 92

which  as you can see doesn't include all the results that it should.

**Reply by sampsyo** [reply]
Aha, since (some of) the numbers have decimal points (like `98.0`), how about trying `float` instead of `int`?

**Reply by bearcatsandor** [reply]
Ah!  Wow. I feel like a doofus because I'm a programmer and should know better. Ok, so float brings up the floats and not the ints, but otherwise works.  I'll go with one or the other and use modify to alter them.

Thank you.
